Allow a custom environment variable prefix to include the environment variable separator

rdodson41 opened the pull request:
Currently, given the following configuration:
```
Config.setup do |config|
  config.use_env = true
  config.env_prefix = 'MY_CONFIG'
  config.env_separator = '_'
  config.env_converter = :downcase
  config.env_parse_values = true
end
```
an environment variable, such as `MY_CONFIG_USERNAME=rdodson41`, will be split into `['MY', 'CONFIG', 'USERNAME']`, and, therefore will fail to match against the prefix `MY_CONFIG`.

This pull request will allow a custom environment variable prefix to include the environment variable separator by splitting the prefix on the separator and, only then, comparing the beginning of the split variable name to the split prefix. As a result, `['MY', 'CONFIG', 'USERNAME']` will successfully match against the prefix `MY_CONFIG`, which would be split into `['MY', 'CONFIG']`.

I don't think that this is a breaking change, as currently, given such a configuration, no environment variables would be merged into the configuration, but I am not completely sure.

pkuczynski commented:
Code climate complain about `Config::Options#split doesn't depend on instance state (maybe move it to another class?)`. At the same time `Config.env_prefix` and `Config.const_name` should be already strings, so no point to convert them.

Could you please inline split method when appropriate?

Other than that looks good! :)

rdodson41 commented:
@pkuczynski Thanks, for the feedback. I reverted the commit in question. Should be good now.

rdodson41 commented:
@pkuczynski I fixed the other two Code Climate issues; the single remaining issue does not originate in this pull request, so I left it unchanged.

With regard to `(Config.env_prefix || Config.const_name).to_s.split(separator)`, I agree that `.to_s` *should* not be necessary, as both of these configuration variables *should* be strings, but I retained `.to_s` so that this method does not raise an error if, for some reason, a user configures one or both variables as symbols.

pkuczynski commented:
Looks good! Thanks!
